### DISCUSSION TITLE
IoT use case | Listeners

### DISCUSSION THREAD

**Discussion by ArjunDhar** [pesos/grofer#140] [Ideas]
**TL;DR**:
- Ability to have real time listeners/callbacks or channels. Usage: IoT (but not limited to it) : Attach to a custom MQTT listener to publish that information
- Listeners can be via API or a piped process (stdout ?); So non GO clients can feed off metric events (Export doesnt seem to cut it)
- Listeners can be defined via CLI or config (can decide)
- Listeners can be activated on thresholds also OR thresholds can be applied to raw listeners that publish the data (same as export formats supported) -- (part of config?!)

Hi, Im exploring the use of this tool. I have not gone through the Source yet so can tell me to RTFM and point me the page :)
I read `export` however, id have to poll export and filter through. A {listener + filter} model would boost the functional scalability of this tool.

thanks
